vicodyn/src/gateway/vicodyn.cpp

```diff
@@ -151,15 +203,53 @@ vicodyn_t::vicodyn_t(context_t& _context, const std::string& _local_uuid, const
             } else {
                 auto it = data.apps.find(app);
                 if(it != data.apps.end()) {
-                    app_result = dynamic_t::empty_object;
                     app_result.as_object()[it->first] = convert_app_services(it->second);
                 }
             }
         });
         return result;
     });
-    d->on<io::vicodyn::info>([]() -> dynamic_t {
-        throw error_t("not implemented");
+    d->on<io::vicodyn::info>([&](std::string app) -> dynamic_t {
```

Comment:
It is possible to define `app` as optional?

Reply:
No, as I understand, cocaine does not support optional args.